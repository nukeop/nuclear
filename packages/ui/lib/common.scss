$bgdarker: #1F212D;
$bgdark: #2C2E3A;
$bgdefault: #424557;
$bglight: #555977;
$bglighter: #6A6F95;

$background: $bgdark;
$background2: $bgdefault;
$background3: $bgdarker;
$lightbg: $bglighter;

$grey: #f8f8f2;
$blue: #6272a4;
$cyan: #8be9fd;
$green: #3CDD65;
$orange: #ffb86c;
$pink: #FC83D3;
$purple: #bd93f9;
$red: #ff5555;
$yellow: #f1fa8c;
$white: #fafafa;
$black: #202020;

$very-short-duration: 0.1s;
$short-duration: 0.3s;
$medium-duration: 0.5s;
$long-duration: 1s;

$mid_screen: 800px;
$small_screen: 600px;

@mixin roundCorners {
  border-radius: 0.25em;
}

@mixin circle {
  border-radius: 9999px;
  overflow: hidden;
}

@mixin transition {
  transition: 0.25s;
}

@mixin center {
  align-items: center;
  justify-content: center;
}

@mixin ellipsis {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@mixin shadow {
  box-shadow: 0 0 38px rgba(0, 0, 0, 0.3), 0 0 12px rgba(0, 0, 0, 0.22) !important;
}

.nuclear {
  font-family: "Lato", sans-serif;
  color: $white;
}

@keyframes fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes fade-out {
  from {
    opacity: 1;
  }
  to {
    opacity: 0;
  }
}
