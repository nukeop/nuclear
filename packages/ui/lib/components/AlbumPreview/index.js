import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import {
  withHandlers,
  withState,
  compose
} from 'recompose';
import { Button } from 'semantic-ui-react';

import { getThumbnail } from '../../utils';
import ContextPopup from '../ContextPopup';
import TrackRow from '../TrackRow';
import artPlaceholder from '../../../resources/media/art_placeholder.png';
import styles from './styles.scss';

const AlbumPreview = ({
  album,
  trackButtons,
  target,
  handleListClick,

  handleAddToQueue,
  handlePlayAll
}) => {
  const thumb = _.defaultTo(getThumbnail(album), artPlaceholder);
  const TrackButtons = trackButtons;

  return (
    <div className={styles.album_preview} >
      <div
        className={styles.album_cover}
        style={{ backgroundImage: `url(${thumb})` }}
      >
        <div className={styles.album_cover_overlay}>
          <Button
            basic
            icon='plus'
            size='huge'
            onClick={handleAddToQueue}
          />
          <Button
            basic
            icon='play'
            size='huge'
            onClick={handlePlayAll}
          />
        </div>
      </div>
      <div
        className={styles.track_list}
        onClick={handleListClick}
      >
        <table>
          <tbody>
            {
              album.tracks.map((track, index) => (
                <ContextPopup
                  trigger={
                    <TrackRow
                      key={index}
                      track={track}
                      mini
                      displayArtist
                      displayAlbum
                      displayTrackNumber
                      displayDuration
                    />
                  }
                  key={index}
                  thumb={thumb}
                  title={track.name}
                  artist={_.get(album, 'artist.name')}
                  target={target}
                >
                  <TrackButtons
                    track={track}
                    withAddToDownloads={false}
                  />
                </ContextPopup>
              ))
            }
          </tbody>
        </table>
      </div>
    </div>
  );
};

AlbumPreview.propTypes = {
  album: PropTypes.shape({
    artist: PropTypes.shape({
      name: PropTypes.string
    }),
    title: PropTypes.string.isRequired,
    tracks: PropTypes.arrayOf(PropTypes.shape({
      album: PropTypes.string,
      artist: PropTypes.shape({
        name: PropTypes.string
      }),
      name: PropTypes.string,
      duration: PropTypes.number,
      position: PropTypes.number,
      playcount: PropTypes.number
    })),
    image: PropTypes.arrayOf(PropTypes.shape({
      '#text': PropTypes.string
    }))
  }),
  trackButtons: PropTypes.object
};

export default compose(
  withState(
    'target', 'setTarget', { x: 0, y: 0 }
  ),
  withHandlers({
    handleListClick: ({ setTarget }) => e => setTarget({ x: e.clientX, y: e.clientY })
  })
)(AlbumPreview);
