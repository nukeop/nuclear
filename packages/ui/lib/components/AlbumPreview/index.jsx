import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Button } from 'semantic-ui-react';

import { getThumbnail } from '../..';
import ContextPopup from '../ContextPopup';
import TrackRow from '../TrackRow';
import artPlaceholder from '../../../resources/media/art_placeholder.png';
import styles from './styles.scss';

const AlbumPreview = ({
  album,
  trackButtons,

  handleAddToQueue,
  handlePlayAll
}) => {
  const thumb = _.defaultTo(getThumbnail(album), artPlaceholder);
  const TrackButtons = trackButtons;

  return (
    <div className={styles.album_preview} >
      <div
        className={styles.album_cover}
        style={{ backgroundImage: `url(${thumb})` }}
      >
        <div className={styles.album_cover_overlay}>
          <Button
            basic
            icon='plus'
            size='huge'
            onClick={handleAddToQueue}
          />
          <Button
            basic
            icon='play'
            size='huge'
            onClick={handlePlayAll}
          />
        </div>
      </div>
      <div
        className={styles.track_list}
      >
        <table>
          <tbody>
            {
              album.tracks.map((track, index) => (
                <ContextPopup
                  trigger={
                    <TrackRow
                      key={index}
                      track={track}
                      mini
                      displayArtist
                      displayAlbum
                      displayTrackNumber
                      displayDuration
                    />
                  }
                  key={index}
                  thumb={thumb}
                  title={track.name}
                  artist={_.get(album, 'artist.name')}
                >
                  <TrackButtons
                    track={track}
                    withAddToDownloads={false}
                  />
                </ContextPopup>
              ))
            }
          </tbody>
        </table>
      </div>
    </div>
  );
};

AlbumPreview.propTypes = {
  album: PropTypes.shape({
    artist: PropTypes.shape({
      name: PropTypes.string
    }),
    title: PropTypes.string.isRequired,
    tracks: PropTypes.arrayOf(PropTypes.shape({
      album: PropTypes.string,
      artist: PropTypes.shape({
        name: PropTypes.string
      }),
      name: PropTypes.string,
      duration: PropTypes.number,
      position: PropTypes.number,
      playcount: PropTypes.number
    })),
    image: PropTypes.arrayOf(PropTypes.shape({
      '#text': PropTypes.string
    }))
  }),
  trackButtons: PropTypes.object
};

export default AlbumPreview;
