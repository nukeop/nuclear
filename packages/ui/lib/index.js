import '../resources/Lato.css';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Card } from './components/Card';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as SmoothImage } from './components/SmoothImage';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as PopupButton } from './components/PopupButton';
export { default as QueueItem } from './components/QueueItem';
export { default as QueuePopup } from './components/QueuePopup';
export { default as Range } from './components/Range';
export { default as Seekbar } from './components/Seekbar';
export { default as ToastContainer } from './components/ToastContainer';
export { default as TrackRow } from './components/TrackRow';
export { default as UserPluginsItem } from './components/UserPluginsItem';

export * from './utils';
