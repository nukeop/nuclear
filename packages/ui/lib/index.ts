import '../resources/Lato.scss';

export { default as AlbumGrid } from './components/AlbumGrid';
export { default as ArtistGrid } from './components/ArtistGrid';
export { default as AlbumPreview } from './components/AlbumPreview';
export { default as Button } from './components/Button';
export { default as Card } from './components/Card';
export { default as Checkbox } from './components/Checkbox';
export { default as ContextPopup } from './components/ContextPopup';
export { default as Cover } from './components/Cover';
export { default as Dropdown } from './components/Dropdown';
export { default as LibraryListTypeToggle } from './components/LibraryListTypeToggle';
export { default as Loader } from './components/Loader';
export { default as MiniPlayer } from './components/MiniPlayer';
export { default as PlayerBar } from './components/PlayerBar';
export { default as PlayerButton } from './components/PlayerButton';
export { default as PopupButton } from './components/PopupButton';
export { default as PopupDropdown } from './components/PopupDropdown';
export { default as QueueItem } from './components/QueueItem';
export { default as Range } from './components/Range';
export { default as SearchBox } from './components/SearchBox';
export { default as Seekbar } from './components/Seekbar';
export { default as SmoothImage } from './components/SmoothImage';
export { default as StreamInfo } from './components/StreamInfo';
export { default as ToastContainer } from './components/ToastContainer';
export { default as Tooltip } from './components/Tooltip';
export { default as TrackPopup } from './components/TrackPopup';
export { default as TrackRow } from './components/TrackRow';
export { default as TrackTable } from './components/TrackTable';
export { default as UserPluginsItem } from './components/UserPluginsItem';
export { default as Visualizer } from './components/Visualizer';
export { default as VisualizerOverlay } from './components/VisualizerOverlay';

export * from './utils';
