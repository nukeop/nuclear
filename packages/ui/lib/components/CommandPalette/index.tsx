import React, { LegacyRef, useCallback, useEffect, useState } from 'react';
import cx from 'classnames';
import { head, isEmpty, take } from 'lodash';
import { Input, SemanticICONS, Transition } from 'semantic-ui-react';
import useOutsideClick from 'react-cool-onclickoutside';

import { CommandPaletteFooter } from './CommandPaletteFooter';
import { CommandPaletteAction } from './CommandPaletteAction';
import { CommandPaletteEmptyState } from './CommandPaletteEmptyState';
import styles from './styles.scss';
import common from '../../common.scss';

export type CommandPaletteAction = {
    id: string;
    name: string;
    shortcut?: string[];
    icon?: SemanticICONS;
    category?: string;
    onUse?: () => void;
}

export type CommandPaletteProps = React.ComponentProps<typeof CommandPaletteFooter> & 
React.ComponentProps<typeof CommandPaletteEmptyState> & {
    searchPlaceholder?: string;
    actions?: CommandPaletteAction[];
    isLoading?: boolean;
    inputValue?: string;
    onInputChange?: (text: string) => void;

    isOpen?: boolean;
    onClose?: () => void;

    inputRef?: LegacyRef<Input>;
};

const CommandPalette: React.FC<CommandPaletteProps> = ({
  isLoading,
  searchPlaceholder = 'What would you like to do?',
  actions=[],
  inputValue,
  onInputChange,
  emptyStateText,
  isOpen,
  onClose,
  inputRef,
  ...footerProps
}) => {
  const [selected, setSelected] = useState<CommandPaletteAction | null>(null);
  const currentIndex = actions.findIndex(action => action.id === selected?.id);
  const displayedActionsLength = Math.min(actions.length, 5);

  const selectNext = useCallback(() => {
    const nextIndex = currentIndex + 1;
    setSelected(actions[nextIndex % displayedActionsLength]);
  }, [actions, currentIndex]);

  const selectPrevious = useCallback(() => {
    const previousIndex = currentIndex - 1 < 0 
      ? displayedActionsLength - 1 
      : currentIndex - 1;
    setSelected(actions[previousIndex]);
  }, [actions, currentIndex]);

  const closeRef = useOutsideClick(onClose);

  useEffect(() => {
    if (isOpen) {
      setSelected(head(actions));
    }
  }, [inputValue]);

  return <Transition
    visible={isOpen}
    animation='fade'
    duration={200}
  >
    <div className={styles.transition_container}>
      <div
        data-testid='command-palette'
        className={cx(
          styles.command_palette_container,
          common.nuclear
        )}
      >
        <div 
          className={styles.command_palette}
          onKeyDown={(e) => {
            if (e.key === 'ArrowDown') {
              selectNext();
            } else if (e.key === 'ArrowUp') {
              selectPrevious();
            } else if (e.key === 'Enter' && selected) {
              selected.onUse();
              setSelected(null);
              onClose();
            } else if (e.key === 'Escape') {
              setSelected(null);
              onClose();
            }
          }}
          ref={closeRef}
        >
          <Input 
            className={styles.input}
            ref={inputRef}
            autoFocus
            iconPosition='left'
            icon={isLoading ? 'circle notch loading' : 'search'}
            size='big'
            placeholder={searchPlaceholder}
            value={inputValue}
            onChange={e => {
              onInputChange(e.target.value);
            }}
          />
          <div 
            className={styles.actions}
          >
            {
              !isLoading &&
                isEmpty(actions)
                ? <CommandPaletteEmptyState emptyStateText={emptyStateText}/>
                : take(actions, 5).map(action => (
                  <CommandPaletteAction 
                    key={action.id} 
                    onSelect={() => setSelected(action)}
                    isSelected={selected === action}
                    onClose={() => {
                      setSelected(null);
                      onClose();
                    }}
                    {...action} 
                  /> 
                ))
            }
          </div>
          <CommandPaletteFooter 
            {...footerProps}
          />
        </div>
      </div>
    </div>
  </Transition>;
};

export default CommandPalette;

