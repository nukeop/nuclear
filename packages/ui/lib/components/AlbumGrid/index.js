import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import _ from 'lodash';
import {Dimmer, Loader } from 'semantic-ui-react';
import { compose, withState, withHandlers } from 'recompose';

import AlbumPreview from '../AlbumPreview';
import Card from '../Card';
import common from '../../common.scss';
import styles from './styles.scss';

export const getThumbnail = album => {
  return _.get(album, 'images[0].uri',
    _.get(album, 'image[0][\'#text\']',
      _.get(album, 'thumb')));
};

const AlbumGrid = ({
  albums,
  onAlbumClick,
  selectedAlbum,
  loading,
  trackButtons,
  withArtistNames,
  withAlbumPreview
}) => (
  <div className={cx(
    common.nuclear,
    styles.album_grid,
    {[styles.loading]: loading}
  )} >
  <div className={styles.album_cards}>
  {
    !loading &&
    !_.isEmpty(albums) &&
    albums.map((album, i) => (
      <Card
      key={i}
      header={album.title}
      content={withArtistNames && _.get(album, 'artist.name')}
      image={getThumbnail(album)}
      onClick={() => onAlbumClick(album)}
      />
    ))
  }
  </div>

  {
    !loading && withAlbumPreview &&
      <>
        <hr />
        <AlbumPreview
          album={selectedAlbum}
          trackButtons={trackButtons}
        />
      </>
    }
    { loading && <Dimmer active><Loader /></Dimmer> }
  </div>
);

AlbumGrid.propTypes = {
  albums: PropTypes.array,
  onAlbumClick: PropTypes.func,
  loading: PropTypes.bool,
  withArtistNames: PropTypes.bool,
  withAlbumPreview: PropTypes.bool,

  trackButtons: PropTypes.arrayOf(PropTypes.node)
};

export default compose(
  withState(
    'selectedAlbum',
    'selectAlbum',
    ({ albums }) => _.head(albums)),
  withHandlers({
    onAlbumClick: ({ selectAlbum }) => album => selectAlbum(album)
  })
)(AlbumGrid);
