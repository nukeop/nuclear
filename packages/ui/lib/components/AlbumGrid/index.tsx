import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import _ from 'lodash';
import { Dimmer, Loader } from 'semantic-ui-react';
import { compose, withState, withHandlers } from 'recompose';

import AlbumPreview from '../AlbumPreview';
import Card from '../Card';
import common from '../../common.scss';
import styles from './styles.scss';
import { getThumbnail, getTrackItem } from '../..';

const AlbumGrid = ({
  albums,
  onAlbumClick,
  removeFavoriteAlbum,
  selectedAlbum,
  loading,
  trackButtons,
  onAddToQueue,
  onPlayAll,
  autoSize,
  withArtistNames,
  withAlbumPreview
}) => (
  <div className={cx(
    common.nuclear,
    styles.album_grid,
    { [styles.loading]: loading },
    { [styles.auto_size]: autoSize && !loading }
  )} >
    <div className={styles.album_cards}>
      {
        !loading &&
          !_.isEmpty(albums) &&
          albums.map((album, i) => (
            <Card
              key={i}
              header={album.title}
              content={withArtistNames && _.get(album, 'artist.name')}
              image={getThumbnail(album)}
              onClick={() => onAlbumClick(album)}
              withMenu={removeFavoriteAlbum ? true : false}
              menuEntries={[
                {
                  type: 'item', props: {
                    children: 'Remove',
                    onClick: () => removeFavoriteAlbum(album)
                  }
                }
              ]}
            />
          ))
      }
    </div>

    {
      !loading && withAlbumPreview &&
        <Fragment>
          <hr />
          <AlbumPreview
            album={selectedAlbum}
            trackButtons={trackButtons}
            handleAddToQueue={onAddToQueue}
            handlePlayAll={onPlayAll}
          />
        </Fragment>
    }
    {loading && <Dimmer active><Loader /></Dimmer>}
  </div>
);

AlbumGrid.propTypes = {
  albums: PropTypes.array,
  streamProviders: PropTypes.array, // eslint-disable-line react/no-unused-prop-types
  onAlbumClick: PropTypes.func,
  removeFavoriteAlbum: PropTypes.func,
  loading: PropTypes.bool,
  addToQueue: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  clearQueue: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  selectSong: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  startPlayback: PropTypes.func, // eslint-disable-line react/no-unused-prop-types
  onAddToQueue: PropTypes.func,
  onPlayAll: PropTypes.func,

  autoSize: PropTypes.bool,
  withArtistNames: PropTypes.bool,
  withAlbumPreview: PropTypes.bool,

  trackButtons: PropTypes.object
};

export default compose(
  withState(
    'selectedAlbum',
    'selectAlbum',
    ({ albums }) => _.head(albums)
  ),
  withHandlers({
    onAlbumClick: ({ onAlbumClick, selectAlbum }) => album => _.isNil(onAlbumClick) ? selectAlbum(album) : onAlbumClick(album),
    onAddToQueue: ({ addToQueue, selectedAlbum }) => () => {
      selectedAlbum.tracks.map(track => addToQueue(getTrackItem(track)));
    }
  }),
  withHandlers({
    onPlayAll: ({ selectSong, startPlayback, onAddToQueue }) => () => {
      onAddToQueue();
      selectSong(0);
      startPlayback();
    }
  })
)(AlbumGrid);
