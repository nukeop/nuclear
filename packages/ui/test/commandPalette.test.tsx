import { CommandPalette } from '../lib';
import { CommandPaletteAction } from '../lib/components/CommandPalette';
import { makeSnapshotTest } from './helpers';

const actions: CommandPaletteAction[] = [
  {
    id: 'play',
    name: 'Play',
    shortcut: ['space'],
    icon: 'play',
    category: 'Playback'
  }, {
    id: 'next',
    name: 'Next',
    shortcut: ['right arrow'],
    icon: 'forward',
    category: 'Playback'
  }, {
    id: 'previous',
    name: 'Previous',
    shortcut: ['left arrow'],
    icon: 'backward',
    category: 'Playback'
  }, {
    id: 'shuffle',
    name: 'Shuffle',
    shortcut: [],
    icon: 'random',
    category: 'Playback'
  }, {
    id: 'repeat',
    name: 'Repeat',
    shortcut: [],
    icon: 'repeat',
    category: 'Playback'
  }, {
    id: 'raise-volume',
    name: 'Raise volume',
    shortcut: [],
    icon: 'volume up',
    category: 'Playback'
  }, {
    id: 'lower-volume',
    name: 'Lower volume',
    shortcut: [],
    icon: 'volume down',
    category: 'Playback'
  }
];

makeSnapshotTest(
  CommandPalette,
  { isLoading: true },
  '(Snapshot) Command palette - loading'
);

makeSnapshotTest(
  CommandPalette,
  {
    actions: [],
    protipText: 'PROTIP:',
    protipContent: 'Narrow down the list of commands by typing in the search box.'
  },
  '(Snapshot) Command palette - empty'
);

makeSnapshotTest(
  CommandPalette,
  {
    actions,
    protipText: 'PROTIP:',
    protipContent: 'Use the keyboard to navigate the command palette.'
  },
  '(Snapshot) Command palette - with actions'
);
