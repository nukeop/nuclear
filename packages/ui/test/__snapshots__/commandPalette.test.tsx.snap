// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Snapshot) Command palette - empty should render correctly 1`] = `
<DocumentFragment>
  <div
    class="fade transition_container visible transition"
  >
    <div
      class="command_palette_container nuclear"
      data-testid="command-palette"
    >
      <div
        class="command_palette"
      >
        <div
          class="ui big left icon input input"
        >
          <input
            placeholder="What would you like to do?"
            type="text"
            value=""
          />
          <i
            aria-hidden="true"
            class="search icon"
          />
        </div>
        <div
          class="actions"
        >
          <div
            class="nuclear command_palette_empty_state"
          >
            <i
              aria-hidden="true"
              class="searchengin massive icon"
            />
            <p />
            <hr />
          </div>
        </div>
        <footer
          class="command_palette_footer"
        >
          <div
            class="protip"
          >
            <div
              class="protip_text"
            >
              PROTIP:
            </div>
            <div
              class="protip_content"
            >
              Narrow down the list of commands by typing in the search box.
            </div>
          </div>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Command palette - loading should render correctly 1`] = `
<DocumentFragment>
  <div
    class="fade transition_container visible transition"
  >
    <div
      class="command_palette_container nuclear"
      data-testid="command-palette"
    >
      <div
        class="command_palette"
      >
        <div
          class="ui big left icon input input"
        >
          <input
            placeholder="What would you like to do?"
            type="text"
            value=""
          />
          <i
            aria-hidden="true"
            class="circle notch loading icon"
          />
        </div>
        <div
          class="actions"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Snapshot) Command palette - with actions should render correctly 1`] = `
<DocumentFragment>
  <div
    class="fade transition_container visible transition"
  >
    <div
      class="command_palette_container nuclear"
      data-testid="command-palette"
    >
      <div
        class="command_palette"
      >
        <div
          class="ui big left icon input input"
        >
          <input
            placeholder="What would you like to do?"
            type="text"
            value=""
          />
          <i
            aria-hidden="true"
            class="search icon"
          />
        </div>
        <div
          class="actions"
        >
          <button
            class="nuclear command_palette_action"
          >
            <div
              class="action_left"
            >
              <i
                aria-hidden="true"
                class="play icon"
              />
              <div
                class="action_name"
              >
                Play
              </div>
            </div>
            <div
              class="action_right"
            >
              <kbd
                class="shortcut"
              >
                space
              </kbd>
            </div>
          </button>
          <button
            class="nuclear command_palette_action"
          >
            <div
              class="action_left"
            >
              <i
                aria-hidden="true"
                class="forward icon"
              />
              <div
                class="action_name"
              >
                Next
              </div>
            </div>
            <div
              class="action_right"
            >
              <kbd
                class="shortcut"
              >
                right arrow
              </kbd>
            </div>
          </button>
          <button
            class="nuclear command_palette_action"
          >
            <div
              class="action_left"
            >
              <i
                aria-hidden="true"
                class="backward icon"
              />
              <div
                class="action_name"
              >
                Previous
              </div>
            </div>
            <div
              class="action_right"
            >
              <kbd
                class="shortcut"
              >
                left arrow
              </kbd>
            </div>
          </button>
          <button
            class="nuclear command_palette_action"
          >
            <div
              class="action_left"
            >
              <i
                aria-hidden="true"
                class="random icon"
              />
              <div
                class="action_name"
              >
                Shuffle
              </div>
            </div>
            <div
              class="action_right"
            />
          </button>
          <button
            class="nuclear command_palette_action"
          >
            <div
              class="action_left"
            >
              <i
                aria-hidden="true"
                class="repeat icon"
              />
              <div
                class="action_name"
              >
                Repeat
              </div>
            </div>
            <div
              class="action_right"
            />
          </button>
        </div>
        <footer
          class="command_palette_footer"
        >
          <div
            class="protip"
          >
            <div
              class="protip_text"
            >
              PROTIP:
            </div>
            <div
              class="protip_content"
            >
              Use the keyboard to navigate the command palette.
            </div>
          </div>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
